src/rust-crypto/backup.ts

```diff
@@ -129,14 +145,10 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
                 await this.enableKeyBackup(backupInfo);
             } else if (activeVersion !== backupInfo.version) {
                 logger.log(`On backup version ${activeVersion} but found version ${backupInfo.version}: switching.`);
+                // This will remove any pending backup request, remove the backup key and reset the backup state of each room key we have.
                 await this.disableKeyBackup();
+                // Enabling will now trigger re-upload of all the keys
```

Comment:
does it? is that in the rust side?

Reply:
`olmMachine.disableBackup()` reset the backed-up status of exiting keys, so next time you ask for the key to backup you will get all of the keys as candidates, so enabling again will re-upload